### DISCUSSION TITLE
i can use a bluetooth device and any python module with Flet?

### DISCUSSION THREAD

**Discussion by 9acca9** [flet-dev/flet#3952] [General]
Hi, sorry the dumb question. I just hear about Flet.
I want to make a device that is pair with a android cellphone. Recieve a variable string from that device in the app, and use Python-Chess (https://python-chess.readthedocs.io/en/latest/) module to interpret, later send that value to lichess api (Berserk module) 

The thing is i have a limitation in which modules i could use?
Thanks!

**Comment by rralen** [comment]
Hello, I don't know if you still have the question but I also had that problem but no Python module is directly compatible with the use of Bluetooth on cell phones so you can use low-process commands. I'll leave you my code if it helps you.  import subprocess
import os
import flet
from flet import *


def main(page: Page):
    page.title = "Bluetooth File Sender"

    you_file = TextField(label="Your file name")
    destination_mac = TextField(label="MAC Address")
    scan_result = Text(weight="bold")

    def scan_devices(e=None):
        try:
            res = subprocess.check_output(["hcitool", "scan"]).decode("utf-8")
            print(res)

            devices = [line.split("\t", maxsplit=1) for line in res.splitlines()[1:]]
            formatted_devices = [
                f"MAC Address: {mac} \t Device: {name}" for mac, name in devices
            ]
            scan_result.value = "\n".join(formatted_devices)
            page.update()

        except Exception as e:
            scan_result.value = f"Error scanning: {str(e)}"
            page.update()

    def send_file_now(e):
        command = f"bluetooth-sendto --device={destination_mac.value} {you_file.value}"
        result = os.system(command)
        if result == 0:
            print("Success! Transfer complete.")
            page.snack_bar = SnackBar(
                Text("Success sending!!", size=10),
                bgcolor="blue"
            )
            page.snack_bar.open = True
            page.update()
        else:
            print("Failed to send file.")
            page.snack_bar = SnackBar(
                Text("Failed to send file!", size=10),
                bgcolor="red"
            )
            page.snack_bar.open = True
            page.update()

    page.add(
        Column([
            Text("Upload file to Bluetooth", size=25, weight="bold"),
            you_file,
            destination_mac,
            ElevatedButton("Scan Devices", bgcolor="gray", color="white", on_click=scan_devices),
            scan_result,
            ElevatedButton("Send Now", bgcolor="blue", color="white", on_click=send_file_now)
        ])
    )

flet.app(target=main)
